src/Microsoft.AspNetCore.SystemWebAdapters/HttpContextWrapper.cs

```diff
@@ -81,5 +81,10 @@ public override IPrincipal User
         public override void RewritePath(string filePath, string pathInfo, string? queryString, bool setClientFilePath) => _context.RewritePath(filePath, pathInfo, queryString, setClientFilePath);
 
         public override void SetSessionStateBehavior(SessionStateBehavior sessionStateBehavior) => _context.SetSessionStateBehavior(sessionStateBehavior);
+
+        public override object GetService(Type serviceType)
+        {
+            return ((HttpContextCore)_context).RequestServices.GetService(serviceType);
```

Comment:
this should delegate to the _context.GetService method. The `HttpContext.GetService()` method could hook into the _context.RequestServices